morphic.js

```diff
@@ -6903,18 +6904,37 @@ MenuMorph.prototype.unselectAllItems = function () {
 MenuMorph.prototype.popup = function (world, pos) {
     this.drawNew();
     this.setPosition(pos);
-    this.addShadow(new Point(2, 2), 80);
-    this.keepWithin(world);
     if (world.activeMenu) {
         world.activeMenu.destroy();
     }
     if (this.items.length < 1 && !this.title) { // don't show empty menus
         return;
     }
+    this.keepVisible(world);
     world.add(this);
-    world.activeMenu = this;
     this.world = world; // optionally enable keyboard support
+    world.activeMenu = this;
+    if (this.scrollMenu != undefined && this.scrollMenu != null) {
+        pos = this.position();
+        var labelHeight = 0;
+        if (this.label != null) {
+            labelHeight = this.label.height() + 2;
+        }
+        this.scrollMenu.world = world;
+        world.add(this.scrollMenu);
+        this.scrollMenu.setPosition(new Point(pos.x + 3, labelHeight + 6));
+        this.bounds.corner.y = Math.round(0.98 * world.height());
+        this.scrollMenu.setWidth(this.width() - 6);
+        this.scrollMenu.setHeight(this.bottom() - this.scrollMenu.top() - 4);
+        for (var i = this.children.length ; i > -1 ; i += -1){
```

Comment:
Unless there's a specific need to iterate backwards, this should be `this.children.forEach()` style loop.